### ISSUE TITLE
Unable to delete user that in a relationship

### ISSUE THREAD

**Issue by TomS-** [directus/directus#9365]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

When trying to delete a user that is part of a `created_by` you get the following error:

```JSON
{
  "errors": [
    {
      "message": "delete from directus_users where id in ('b8d7cc20-5b5a-4c71-846b-8a9e0341bb25') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (sql_quizcdn.user_answers, CONSTRAINT user_answers_user_foreign FOREIGN KEY (user) REFERENCES directus_users (id))",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

I would expect it to nullify the field as is the default for settings for `created_by`. 

### To Reproduce

1. Create a Collection with a Created_By. 
2. Create some items by that user.  
3. Try to remove that user.

### What version of Directus are you using?

RC.100

### What version of Node.js are you using?

14.17.6

### What database are you using?

MySQL 5.7

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

Digital Ocean

**Comment by rijkvanzanten** [comment]
Duplicate of https://github.com/directus/directus/issues/8650, fixed in https://github.com/directus/directus/pull/9305 🙂 

TL;DR: By default, Directus didn't set any foreign key constraint triggers for the "optional system fields", meaning that they prevent the user from being deleted if their ID is associated to a related item. You could either manually remove the related items, or change the foreign key constraint to SET NULL or CASCADE to automate it when the user is deleted. The referenced PR changed the default from NO ACTION to SET NULL to prevent this confusion 👍🏻

**Comment by TomS-** [comment]
Hey @rijkvanzanten 

Was this expected to be fixed in RC.101? I'm still having the issue.
![image](https://user-images.githubusercontent.com/4595662/140299515-8296f190-b1a1-4f2a-b8c7-f3ec3720ecd3.png)

Deleting gives me the error
```
{
  "errors": [
    {
      "message": "delete from `directus_users` where `id` in ('b8d7cc20-5b5a-4c71-846b-8a9e0341bb25') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`sql_quizcdn`.`user_answers`, CONSTRAINT `user_answers_user_foreign` FOREIGN KEY (`user`) REFERENCES `directus_users` (`id`))",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

**Comment by rijkvanzanten** [comment]
That fix didn't change any existing fields / setups; it only sets better defaults for newly created collections 👍🏻
